pyodide-build/pyodide_build/out_of_tree/venv.py

```diff
@@ -0,0 +1,287 @@
+    )
+    sys.exit(1)
+
+
+def create_pip_conf(venv_root: Path) -> None:
+    """Create pip.conf file in venv root
+
+    This file adds a few options that will always be used by pip install.
+    """
+    if in_xbuild_env():
+        # In the xbuild_env, we don't have the packages locally. We will include
+        # in the xbuildenv a PEP 503 index for the vendored Pyodide packages
+        # https://peps.python.org/pep-0503/
+        repo = f'extra-index-url={get_pyodide_root()/"dist/pypi_index"}'
+    else:
+        # In the Pyodide development environment, the Pyodide dist directory
+        # should contain the needed wheels. find-links
+        repo = f'find-links={get_pyodide_root()/"dist"}'
+
+    (venv_root / "pip.conf").write_text(
+        dedent(
+            f"""
+            [install]
+            only-binary=:all:
+            {repo}
+            """
+        )
+    )
+
+
+def get_pip_monkeypatch(venv_bin: Path) -> str:
+    """Monkey patch pip's environment to show info about Pyodide's environment.
+
+    The code returned is injected at the beginning of the pip script.
+    """
+    result = subprocess.run(
+        [
+            venv_bin / "python",
+            "-c",
+            dedent(
+                """
+                import os, sys, sysconfig, platform
+                print([
+                    os.name,
+                    sys.platform,
+                    sys.implementation._multiarch,
+                    sysconfig.get_platform()
+                ])
+                """
+            ),
+        ],
+        capture_output=True,
+        encoding="utf8",
+    )
+    check_result(result, "ERROR: failed to invoke Pyodide")
+    platform_data = result.stdout
+    sysconfigdata_dir = Path(get_make_flag("TARGETINSTALLDIR")) / "sysconfigdata"
+
+    return dedent(
+        f"""\
+        import os
+        import sys
+        os_name, sys_platform, multiarch, host_platform = {platform_data}
+        os.name = os_name
+        sys.platform = sys_platform
+        sys.implementation._multiarch = multiarch
+        os.environ["_PYTHON_HOST_PLATFORM"] = host_platform
+        os.environ["_PYTHON_SYSCONFIGDATA_NAME"] = f'_sysconfigdata_{{sys.abiflags}}_{{sys.platform}}_{{sys.implementation._multiarch}}'
+        sys.path.append("{sysconfigdata_dir}")
+        """
+    )
+
+
+def create_pip_script(venv_bin):
+    """Create pip and write it into the virtualenv bin folder."""
+    # pip needs to run in the host Python not in Pyodide, so we'll use the host
+    # Python in the shebang. Use whichever Python was used to invoke
+    # pyodide venv.
+    host_python_path = venv_bin / "python3.10-host"
+    (venv_bin / "pip").write_text(
+        # Other than the shebang and the monkey patch, this is exactly what
+        # normal pip looks like.
+        dedent(
+            f"""
+            #!{host_python_path}
+            # -*- coding: utf-8 -*-
```

Comment:
Is this still required? IIRC this was a Python 2 stuff.

Reply:
Probably not, I just copied from pip.

Reply:
Removed it:
https://stackoverflow.com/questions/14083111/should-i-use-encoding-declaration-in-python-3